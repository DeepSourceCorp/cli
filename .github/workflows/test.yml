name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
