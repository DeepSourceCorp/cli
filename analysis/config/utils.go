package config

import (
	"log"
	"os"
	"path"
	"path/filepath"
	"strings"

	"github.com/bmatcuk/doublestar/v4"
)

/* Walks the `CODE_PATH` directory and returns all the files other than the ones present
 * in the .git folder for analysis in the form of a string array */
func readAllFiles(codePath string) ([]string, error) {
	fileCount := 0

	allFiles := make([]string, 0)

	err := filepath.Walk(codePath,
		func(path string, fileInfo os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			fileCount++

			/* Check the following before appending to the list of files:
			 * - Should not be a directory
			 * - The walked file should not be present in .git folder
			 * TODO: Check why fileCount != 1 condition is used here */
			if !fileInfo.IsDir() || !strings.HasPrefix(path, filepath.Join(codePath, ".git")) && fileCount != 1 {
				allFiles = append(allFiles, path)
			}
			return nil
		})
	if err != nil {
		return allFiles, err
	}
	return allFiles, nil
}

// Returns the slice of files matching certain glob patterns
func (r *AnalysisRun) getMatchingFiles(patterns []string) ([]string, error) {
	matchedFiles := make([]string, 0)

	// Return all the files if no exclude_patterns are configured
	if len(patterns) == 0 {
		return matchedFiles, nil
	}

	for _, file := range r.AnalysisFiles {
		for i := range patterns {
			match, err := doublestar.Match(path.Join(r.CodePath, patterns[i]), file)
			if err != nil {
				log.Printf("Failed to match the file %s with pattern %s\n", file, patterns[i])
			}
			if match {
				matchedFiles = append(matchedFiles, file)
				break
			}
		}
	}
	return matchedFiles, nil
}

// Filters the analysis files and removes the files matching the exclude_patterns from them
// TODO: Improve the logic here
func (r *AnalysisRun) filterAnalysisFiles() {
	filteredFiles := []string{}
	for _, file := range r.AnalysisFiles {
		for _, excludedFile := range r.ExcludedFiles {
			if file == excludedFile {
				break
			}
			filteredFiles = append(filteredFiles, file)
		}
	}
	r.AnalysisFiles = filteredFiles
}
