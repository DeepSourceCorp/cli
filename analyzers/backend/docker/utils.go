package docker

import (
	"bufio"
	"encoding/json"
	"errors"
	"fmt"
	"io"
	"strings"
)

type ErrorLine struct {
	Error       string      `json:"error"`
	ErrorDetail ErrorDetail `json:"errorDetail"`
}

type ErrorDetail struct {
	Message string `json:"message"`
}

type DockerBuildResponse struct {
	Stream string `json:"stream"`
}

func CheckBuildResponse(rd io.Reader, showAllLogs bool) error {
	var lastLine []byte
	count := 0
	var currentStream string

	scanner := bufio.NewScanner(rd)
	for scanner.Scan() {
		lastLine = scanner.Bytes()
		d := &DockerBuildResponse{}
		err := json.Unmarshal(lastLine, d)
		if err != nil {
			return err
		}
		if d.Stream == "" || d.Stream == "\n" || strings.Contains(d.Stream, "--->") || strings.TrimSuffix(d.Stream, "\n") == currentStream {
			continue
		}
		currentStream = strings.TrimSuffix(d.Stream, "\n")
		if showAllLogs {
			fmt.Println(currentStream)
		}
		count++
	}

	errLine := &ErrorLine{}
	json.Unmarshal([]byte(lastLine), errLine)
	if errLine.Error != "" {
		return errors.New(errLine.Error)
	}
	return scanner.Err()
}


