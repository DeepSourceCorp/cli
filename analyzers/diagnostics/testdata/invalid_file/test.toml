name = "2do-Checker"
