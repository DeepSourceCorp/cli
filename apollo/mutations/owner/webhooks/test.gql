mutation testWebhook($webhookId: ID!) {
  testWebhook(input: { webhookId: $webhookId }) {
    ok
  }
}
