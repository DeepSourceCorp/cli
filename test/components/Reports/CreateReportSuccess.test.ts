import { render } from '@testing-library/vue'
import { VueConstructor } from 'vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'

import VTooltip from 'v-tooltip'
import { CreateReportSuccess } from '~/components/Reports'

import { cartesian, generateBooleanProps, generateStringProps } from '~/test/utils'
import { mocksGenerator } from '~/test/mocks'

interface CreateReportSuccessT extends Vue {
  formattedUrl: string
  reportDetails: string
}

describe('[[ CreateReportSuccess ]]', () => {
  let localVue: VueConstructor<Vue>

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VTooltip)
  })

  const baseProps = {
    reportId: '123456'
  }

  const stubs = {
    ZAlert: true,
    ZButton: true,
    ZIcon: true,
    ZInput: true,
    ZDialogGeneric: true,
    CopyButton: true
  }

  const mockData = mocksGenerator()

  test('renders CreateReportSuccess with all prop options', () => {
    const editModeValues = generateBooleanProps('editMode')
    const passwordValues = generateStringProps('password', ['deepsource'])

    cartesian(editModeValues, passwordValues).forEach((propCombination) => {
      const { html } = render(CreateReportSuccess, {
        props: {
          ...baseProps,
          ...propCombination
        },
        stubs,
        mocks: mockData,
        localVue
      })

      expect(html()).toMatchSnapshot(JSON.stringify(propCombination))
    })
  })

  describe('[[ CreateReportSuccess getters ]]', () => {
    describe('[[ formattedUrl ]]', () => {
      test('Test for formattedUrl getter when password not available', () => {
        const wrapper = shallowMount(CreateReportSuccess, {
          stubs,
          propsData: baseProps,
          mocks: mockData,
          localVue
        })

        const vm = wrapper.vm as CreateReportSuccessT

        const { reportId } = baseProps

        const url = `${window.location.origin}/report/${reportId}`

        // Test formattedUrl returns correct value
        expect(vm.formattedUrl).toBe(url)
      })
    })

    describe('[[ reportDetails ]]', () => {
      test('Test for reportDetails getter when password not available', () => {
        const wrapper = shallowMount(CreateReportSuccess, {
          stubs,
          propsData: baseProps,
          mocks: mockData,
          localVue
        })

        const vm = wrapper.vm as CreateReportSuccessT

        const { formattedUrl } = vm

        const details = `View report -\nLink - ${formattedUrl}`

        // Test reportDetails returns correct value
        expect(vm.reportDetails).toBe(details)
      })

      test('Test for reportDetails getter when password is available', () => {
        const props = {
          ...baseProps,
          password: 'deepsource'
        }
        const wrapper = shallowMount(CreateReportSuccess, {
          stubs,
          propsData: props,
          mocks: mockData,
          localVue
        })

        const vm = wrapper.vm as CreateReportSuccessT

        const { formattedUrl } = vm
        const { password } = props

        const details = `View report -\nLink - ${formattedUrl}\nPassword - ${password}`

        // Test reportDetails returns correct value
        expect(vm.reportDetails).toBe(details)
      })
    })
  })
})
