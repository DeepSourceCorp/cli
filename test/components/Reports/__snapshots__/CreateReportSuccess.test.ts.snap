// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"editMode":false} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"editMode":null} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"editMode":true} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully updated
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":"deepsource","editMode":false} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <zalert-stub type=\\"warning\\" class=\\"pt-2.5\\">
      <p class=\\"inline-flex items-start gap-x-2.5 leading-6 tracking-snug\\">
        <zicon-stub icon=\\"solid-alert-circle\\" size=\\"base\\" position=\\"\\" color=\\"honey-500\\"></zicon-stub>
        Please make sure you've copied the password. You would not be able to see it again!
      </p>
    </zalert-stub> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456?password=deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label> <label for=\\"report-password\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Password</p>
      <zinput-stub name=\\"\\" value=\\"deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report password\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-password\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456?password=deepsource
Password - deepsource\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":"deepsource","editMode":null} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <zalert-stub type=\\"warning\\" class=\\"pt-2.5\\">
      <p class=\\"inline-flex items-start gap-x-2.5 leading-6 tracking-snug\\">
        <zicon-stub icon=\\"solid-alert-circle\\" size=\\"base\\" position=\\"\\" color=\\"honey-500\\"></zicon-stub>
        Please make sure you've copied the password. You would not be able to see it again!
      </p>
    </zalert-stub> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456?password=deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label> <label for=\\"report-password\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Password</p>
      <zinput-stub name=\\"\\" value=\\"deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report password\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-password\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456?password=deepsource
Password - deepsource\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":"deepsource","editMode":true} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully updated
    </h3>
    <zalert-stub type=\\"warning\\" class=\\"pt-2.5\\">
      <p class=\\"inline-flex items-start gap-x-2.5 leading-6 tracking-snug\\">
        <zicon-stub icon=\\"solid-alert-circle\\" size=\\"base\\" position=\\"\\" color=\\"honey-500\\"></zicon-stub>
        Please make sure you've copied the password. You would not be able to see it again!
      </p>
    </zalert-stub> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456?password=deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label> <label for=\\"report-password\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Password</p>
      <zinput-stub name=\\"\\" value=\\"deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report password\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-password\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456?password=deepsource
Password - deepsource\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":"deepsource"} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <zalert-stub type=\\"warning\\" class=\\"pt-2.5\\">
      <p class=\\"inline-flex items-start gap-x-2.5 leading-6 tracking-snug\\">
        <zicon-stub icon=\\"solid-alert-circle\\" size=\\"base\\" position=\\"\\" color=\\"honey-500\\"></zicon-stub>
        Please make sure you've copied the password. You would not be able to see it again!
      </p>
    </zalert-stub> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456?password=deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label> <label for=\\"report-password\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Password</p>
      <zinput-stub name=\\"\\" value=\\"deepsource\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report password\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-password\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456?password=deepsource
Password - deepsource\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":null,"editMode":false} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":null,"editMode":null} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":null,"editMode":true} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully updated
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {"password":null} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;

exports[`[[ CreateReportSuccess ]] renders CreateReportSuccess with all prop options: {} 1`] = `
"<zdialoggeneric-stub>
  <div class=\\"p-4 flex flex-col gap-y-4 w-98\\">
    <h3 class=\\"text-sm font-medium text-vanilla-100 leading-6\\">
      Report has been successfully created
    </h3>
    <!----> <label for=\\"report-link\\" class=\\"space-y-1.5\\">
      <p class=\\"text-xs font-normal text-vanilla-100 leading-5\\">Report URL</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/123456\\" label=\\"\\" type=\\"text\\" placeholder=\\"Report URL\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\" class=\\"border border-ink-200\\"></zinput-stub>
    </label>
    <!---->
    <div>
      <copybutton-stub button-type=\\"link\\" color=\\"vanilla-400\\" icon-color=\\"current\\" icon-size=\\"x-small\\" label=\\"Copy report details\\" succes-label=\\"Report details copied\\" value=\\"View report -
Link - https://deepsource.io/report/123456\\" class=\\"float-left hover:text-vanilla-100 pl-1 hover:no-underline\\"></copybutton-stub>
      <zbutton-stub color=\\"\\" iconcolor=\\"\\" label=\\"Done\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"primary\\" type=\\"button\\" size=\\"small\\" to=\\"\\" icon=\\"check\\" class=\\"float-right\\"></zbutton-stub>
    </div>
  </div>
</zdialoggeneric-stub>"
`;
